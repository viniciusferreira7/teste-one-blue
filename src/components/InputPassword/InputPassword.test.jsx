import { render, screen } from '@testing-library/react';
import { InputPassword } from '.';

describe('<InputPassword/>', () => {
  it('should render text in the placeholder', () => {
    render(<InputPassword text="Senha" />);
    expect(screen.getByPlaceholderText('Senha')).toBeInTheDocument();
  });

  it('should render text in the placeholder', () => {
    render(<InputPassword name="password" text="Senha" />);
    expect(screen.getByPlaceholderText('Senha')).toHaveAttribute(
      'name',
      'password',
    );
  });

  it('should have attribute required', () => {
    render(<InputPassword name="password" text="Senha" />);
    expect(screen.getByPlaceholderText('Senha')).toHaveAttribute('required');
  });
  it('should correct style', () => {
    render(<InputPassword name="password" text="Senha" />);
    expect(screen.getByPlaceholderText('Senha')).toHaveStyle(
      'border: 0.2rem solid #5d0cff',
    );
  });

  it('should match snapshot', () => {
    render(<InputPassword name="password" text="Senha" />);
    expect(screen.getByPlaceholderText('Senha')).toMatchSnapshot();
  });
});
