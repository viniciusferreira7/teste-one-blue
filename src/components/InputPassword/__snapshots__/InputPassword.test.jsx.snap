// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputPassword/> should match snapshot 1`] = `
<input
  class="sc-bczRLJ drTqQa"
  name="password"
  placeholder="Senha"
  required=""
  type="password"
/>
`;
